src/main/java/com/scopely/mapper/JsonDynamoMapper.java

```diff
@@ -113,6 +117,41 @@ public JsonDynamoMapper(AmazonDynamoDB amazonDynamoDB, ObjectMapper objectMapper
         return putItem(serialized, tableName, autoGeneratedAttributes);
     }
 
+    public <T> void saveAll(Class<T> clazz, List<T> items) throws MappingException {
+        Class<?> annotatedClazz = findAnnotatedClass(clazz, DynamoDBTable.class);
+        if (annotatedClazz == null) {
+            throw new MappingException("Could not find annotated interface for provided class " + clazz);
+        }
+
+        String tableName = tableName(clazz);
+        List<String> autoGeneratedAttributes = autoGeneratedAttributes(clazz);
+
+        List<WriteRequest> writeRequests = new ArrayList<>();
+        Map<String, List<WriteRequest>> requestItems = new HashMap<>();
+        requestItems.put(tableName, writeRequests);
+        BatchWriteItemRequest writeRequest = new BatchWriteItemRequest(requestItems);
```

Comment:
should these 3 lines be *after* the for loop? Otherwise you're initializing `requestItems` with `writeRequests` before adding items to `writeRequests`.

Reply:
But this is a reference type, so I'm adding a reference to the collection, so making changes to the referenced collection is fine, isn't it?

Reply:
Oh, I guess I didn't realize that was the case. I think its clearer to construct the entity when you need it, though, which would be after the loop. I guess its not a huge deal though if tests are passing.

Reply:
Moved these lines after the loop for clarify